ScienceDirect.js

```diff
@@ -387,47 +345,42 @@ function processRIS(doc, text) {
 	translator.translate();
 }
 
-function scrapeByISBN(doc) {
-	var isbn = getISBN(doc);
-	var translator = Zotero.loadTranslator("search");
-	translator.setTranslator("c73a4a8c-3ef1-4ec8-8229-7531ee384cc4");
-	translator.setSearch({
-		ISBN: isbn
-	});
-	translator.translate();
-}
 
 function getArticleList(doc) {
+	/* eslint-disable no-multi-str */
 	return ZU.xpath(doc,
 		'(//table[@class="resultRow"]/tbody/tr/td[2]/a\
 			|//table[@class="resultRow"]/tbody/tr/td[2]/h3/a\
 			|//td[@class="nonSerialResultsList"]/h3/a\
 			|//div[@id="bodyMainResults"]//li[contains(@class,"title")]//a\
-			|//h2/a[contains(@class, "result-list-title-link")]\
+			|//h2//a[contains(@class, "result-list-title-link")]\
 			|//ol[contains(@class, "article-list") or contains(@class, "article-list-items")]//a[contains(@class, "article-content-title")]\
 			|//li[contains(@class, "list-chapter")]//h2//a\
-		)\[not(contains(text(),"PDF (") or contains(text(), "Related Articles"))]');
+			|//h4[contains(@class, "chapter-title")]/a\
+		)[not(contains(text(),"PDF (") or contains(text(), "Related Articles"))]');
+	/* eslint-enable no-multi-str */
```

Comment:
@dstillman Is that multi line string here okay like this? Or do you suggest something else instead?

Reply:
An XPath like this, where whitespace isn't meaningful, is a rare case where not using string concatenation is reasonable, but it should use ES6 template strings instead:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Multi-line_strings

(I don't think the backslash method was ever even technically part of ECMAScript.)

Reply:
Okay, but I guess that I will the rewrite it slightly different as:
```js
	let articlePaths = [
		'//table[@class="resultRow"]/tbody/tr/td[2]/a',
		'//table[@class="resultRow"]/tbody/tr/td[2]/h3/a',
		'//td[@class="nonSerialResultsList"]/h3/a',
		'//div[@id="bodyMainResults"]//li[contains(@class,"title")]//a',
		'//h2//a[contains(@class, "result-list-title-link")]',
		'//ol[contains(@class, "article-list") or contains(@class, "article-list-items")]//a[contains(@class, "article-content-title")]',
		'//li[contains(@class, "list-chapter")]//h2//a',
		'//h4[contains(@class, "chapter-title")]/a'
	];
	return ZU.xpath(doc, '('
		+ articlePaths.join('|')
		+ ')[not(contains(text(),"PDF (") or contains(text(), "Related Articles"))]'
	);
```
That is easier to read and satisfying the linting rules.

(For the backslash method see https://eslint.org/docs/rules/no-multi-str)